cloudify_rest_client/resources.py

```diff
@@ -69,6 +110,73 @@ def upload_deployment_workdir(self, deployment_id: str, src_dir: str):
                 data=data,
             )
 
+    def _upload_deployment_workdir_files(self, uri: str, src_dir: str):
+        manager_files = self._fetch_manager_directory_index(uri)
+        local_files = self._generate_directory_metadata(src_dir)
+
+        self._save_local_index(src_dir, local_files)
+
+        for file_path, file_mtime in local_files.items():
+            if file_path not in manager_files or \
+                    manager_files[file_path] != file_mtime:
+                self._upload_single_file(
+                    f'{uri}{file_path}',
+                    os.path.join(src_dir, file_path),
+                )
+
+    def _upload_single_file(self, uri: str, file_path: str, extract=False):
```

Comment:
we'll eventually need a function like this, but in public API; the signature would then need to be something like `def upload(self, deployment_id, target_file_path, src_file)`

Maybe we can do it already.